core/domain/skill_services.py

```diff
@@ -230,6 +235,251 @@ def get_all_skill_summaries():
+                might be None.
+            more: bool. If True, there are (probably) more results after this
+                batch. If False, there are no further results after this batch.
+    """
+    augmented_skill_summaries = []
+
+    augmented_skill_summaries_batch, new_urlsafe_start_cursor, more = (
+        _get_augmented_skill_summaries_in_batches(
+            num_skills_to_fetch, urlsafe_start_cursor, sort_by))
+    filtered_augmented_skill_summaries = _filter_skills_by_status(
+        augmented_skill_summaries_batch, status)
+    filtered_augmented_skill_summaries = _filter_skills_by_classroom(
+        filtered_augmented_skill_summaries, classroom_name)
+
+    filtered_augmented_skill_summaries = _filter_skills_by_keywords(
+        filtered_augmented_skill_summaries, keywords)
+    augmented_skill_summaries.extend(filtered_augmented_skill_summaries)
+
+    while (len(augmented_skill_summaries) < num_skills_to_fetch
+           and more and new_urlsafe_start_cursor):
+        augmented_skill_summaries_batch, new_urlsafe_start_cursor, more = (
+            _get_augmented_skill_summaries_in_batches(
+                num_skills_to_fetch, new_urlsafe_start_cursor, sort_by))
+
+        filtered_augmented_skill_summaries = _filter_skills_by_status(
+            augmented_skill_summaries_batch, status)
+        filtered_augmented_skill_summaries = _filter_skills_by_classroom(
+            filtered_augmented_skill_summaries, classroom_name)
+        filtered_augmented_skill_summaries = _filter_skills_by_keywords(
+            filtered_augmented_skill_summaries, keywords)
+        augmented_skill_summaries.extend(filtered_augmented_skill_summaries)
+
+    return augmented_skill_summaries, new_urlsafe_start_cursor, more
+
+
+def _get_augmented_skill_summaries_in_batches(
+        num_skills_to_fetch, urlsafe_start_cursor, sort_by):
+    """Returns all the Augmented skill summaries after attaching
+    topic and classroom.
+
+    Returns:
+        3-tuple(augmented_skill_summaries, urlsafe_start_cursor, more). where:
+            augmented_skill_summaries: list(AugmentedSkillSummary). The list of
+                skill summaries.
+            urlsafe_start_cursor: str or None. A query cursor pointing to the
+                next batch of results. If there are no more results, this might
+                be None.
+            more: bool. If True, there are (probably) more results after this
+                batch. If False, there are no further results after this batch.
+    """
+    skill_summaries, new_urlsafe_start_cursor, more = (
+        _get_skill_summaries_in_batches(
+            num_skills_to_fetch, urlsafe_start_cursor, sort_by))
+
+    assigned_skill_ids = {}
+
+    all_topic_models = topic_models.TopicModel.get_all()
+    all_topic_ids = [topic_model.id for topic_model in all_topic_models]
+    all_topics = topic_fetchers.get_topics_by_ids(all_topic_ids)
```

Comment:
Wait, this would send 2 fetch requests to the DB whereas only one is required, right? Can't you just use `get_topic_from_model` in a loop here and use the models to get the domain objects?

Reply:
Fixed.